packages/near-swap-find-provider/lib/NearSwapFindProvider.ts

```diff
@@ -0,0 +1,176 @@
+    super({
+      baseURL: url,
+      responseType: 'text',
+      transformResponse: undefined // https://github.com/axios/axios/issues/907,
+    })
+  }
+
+  normalizeTransactionResponse(tx: near.NearScraperSwap): near.NearSwapTransaction {
+    const normalizedTx = {
+      hash: `${tx.hash}_${tx.signer_id}`,
+      blockHash: tx.block_hash,
+      sender: tx.signer_id,
+      receiver: tx.receiver_id,
+      rawHash: tx.hash
+    } as { [key: string]: any }
+
+    switch (tx.action_kind) {
+      case 'DEPLOY_CONTRACT': {
+        const code = toBase64(tx.args.code_sha256)
+        normalizedTx.code = code
+        break
+      }
+
+      case 'TRANSFER': {
+        const value = tx.args.deposit
+        normalizedTx.value = value
+        break
+      }
+
+      case 'FUNCTION_CALL': {
+        const method = tx.args.method_name
+        const args = fromBase64(tx.args.args_base64)
+
+        switch (method) {
+          case 'init': {
+            normalizedTx.swap = {
+              method,
+              secretHash: fromBase64(args.secretHash, 'hex'),
+              expiration: fromNearTimestamp(args.expiration),
+              recipient: args.buyer
+            }
+            break
+          }
+
+          case 'claim': {
+            normalizedTx.swap = {
+              method,
+              secret: fromBase64(args.secret, 'hex')
+            }
+
+            break
+          }
+
+          case 'refund': {
+            normalizedTx.swap = { method }
+            break
+          }
+
+          default: {
+            normalizedTx._raw = { method, ...args }
+            break
+          }
+        }
+        break
+      }
+
+      default: {
+        break
+      }
+    }
+    return normalizedTx as near.NearSwapTransaction
+  }
+
+  async findAddressTransaction(
+    address: string,
+    predicate: (tx: near.NearSwapTransaction) => boolean,
+    limit = 1024
+  ): Promise<Transaction<near.NearSwapTransaction>> {
+    let offset = this.getCurrentTimeInNs()
+
+    for (let page = 1; ; page++) {
+      const transactions = (await this.nodeGet(
+        `account/${address}/activity?offset=${offset}&limit=${limit}`
+      )) as near.NearScraperSwap[]
+
+      if (transactions.length === 0) {
+        return
+      }
+
+      const normalizedTransactions = {} as { [key: string]: near.NearSwapTransaction }
+
+      for (const tx of transactions) {
+        normalizedTransactions[tx.hash] = {
+          ...normalizedTransactions[tx.hash],
+          ...this.normalizeTransactionResponse(tx)
+        }
+      }
+
+      const tx = Object.values(normalizedTransactions).find(predicate) as near.NearSwapTransaction
+
+      if (tx) {
+        const currentHeight = await this.getMethod('getBlockHeight')()
+        const txBlockHeight = await this.getMethod('getBlockHeight')(tx.blockHash)
+        tx.confirmations = currentHeight - txBlockHeight
+        return { ...tx, secret: tx.swap.secret, _raw: tx } as Transaction<near.NearSwapTransaction>
```

Comment:
Might be wrong here, but doesn't casting like this with `as` cause the type checking to be ignored? 
i'm worried some fields will be incorrect by doing it like this

Reply:
Yeah, that casting was redundant.